Source/WebCore/rendering/style/StyleCustomPropertyData.cpp

```diff
@@ -81,6 +82,9 @@ void StyleCustomPropertyData::set(const AtomString& name, Ref<const CSSCustomPro
         return !existing || !existing->equals(value);
     }());
 
+    if (value->isAnimatable())
```

Comment:
Should you add a `!m_mayHaveAnimatableProperties` check there?

```suggestion
    if (!m_mayHaveAnimatableProperties && value->isAnimatable())
```